packages/dataclass-editor/src/detail/field/InputFieldWithTypeBrowser.tsx

```diff
@@ -0,0 +1,89 @@
+import {
+  BasicField,
+  BasicInput,
+  BrowsersView,
+  type Browser,
+  Button,
+  Dialog,
+  DialogContent,
+  DialogTrigger,
+  InputGroup,
+  type BrowserNode,
+  useBrowser,
+  BasicCheckbox,
+  type MessageData
+} from '@axonivy/ui-components';
+import { IvyIcons } from '@axonivy/ui-icons';
+import { useEffect, useState } from 'react';
+import { typeData } from '../../data/type-data';
+import { useAppContext } from '../../context/AppContext';
+import { useMeta } from '../../context/useMeta';
+import type { DataclassType } from '../../protocol/types';
+
+export type InputFieldProps = {
+  value: string;
+  onChange: (value: string) => void;
+  onBlur?: () => void;
+  message?: MessageData;
+};
+
+export const InputFieldWithTypeBrowser = ({ value, onChange, onBlur, message }: InputFieldProps) => {
+  const [open, setOpen] = useState(false);
+  const { context } = useAppContext();
+  const [types, setTypes] = useState<Array<BrowserNode<DataclassType>>>([]);
+  const [typeAsList, setTypeAsList] = useState<boolean>(false);
+  const dataClasses = useMeta('meta/scripting/dataClasses', context, []).data;
+  const ivyTypes = useMeta('meta/scripting/ivyTypes', undefined, []).data;
+
+  useEffect(() => {
+    setTypes(typeData(dataClasses, ivyTypes));
+  }, [dataClasses, ivyTypes]);
```

Comment:
instead of this useEffect with state, we should better use a useMemo hook

```tsx
const types = useMemo(() => typeData(dataClasses, ivyTypes), [dataClasses, ivyTypes]);
```